sn_client/src/audit/dag_crawling.rs

```diff
@@ -193,14 +194,27 @@ impl Client {
         let mut utxos = BTreeSet::new();
 
         // get first spend
-        let first_spend = match self.crawl_spend(spend_addr).await {
-            InternalGetNetworkSpend::Spend(s) => *s,
-            InternalGetNetworkSpend::DoubleSpend(s1, s2) => {
+        let mut txs_to_follow = match self.crawl_spend(spend_addr).await {
+            InternalGetNetworkSpend::Spend(spend) => {
+                let spend = *spend;
+                let txs = BTreeSet::from_iter([spend.spend.spent_tx.clone()]);
+
                 spend_processing
-                    .send(*s2)
+                    .send(spend)
                     .await
                     .map_err(|e| WalletError::SpendProcessing(e.to_string()))?;
-                *s1
+                txs
+            }
+            InternalGetNetworkSpend::DoubleSpend(spends) => {
+                let mut txs = BTreeSet::new();
+                for spend in spends.into_iter() {
+                    txs.insert(spend.spend.spent_tx.clone());
+                    spend_processing
```

Comment:
Earlier only the first spend was sent for spend_processing, but here we do it for all the spends, is this alright?